src/pages/fieldManagement/settings/defaultFieldComponents/RelationshipEditorCore.jsx

```diff
@@ -0,0 +1,239 @@
+  return allRelationships;
+}
+
+function addRelationshipToCategory(
+  allRelationships,
+  currentCategory,
+  newRelationship,
+) {
+  const currentRelationships = get(
+    allRelationships,
+    currentCategory,
+    [],
+  );
+  let newRelationships = currentRelationships;
+  if (!currentRelationships.includes(newRelationship))
+    newRelationships = [...currentRelationships, newRelationship];
+  allRelationships[currentCategory] = newRelationships;
+  return allRelationships;
+}
+
+function renderSingleRelationship(
+  allRelationships,
+  currentCategory,
+  relationship,
+  onChange,
+) {
+  return (
+    <div style={{ marginLeft: 32, marginTop: 10 }} key={relationship}>
+      <TextInput
+        width={240}
+        schema={relationship}
+        onChange={newRelationshipName => {
+          onChange(
+            updateRelationshipInCategory(
+              allRelationships,
+              currentCategory,
+              relationship,
+              newRelationshipName,
+            ),
+          );
+        }}
+        value={relationship}
+        autoFocus
+        InputProps={{
+          endAdornment: (
+            <InputAdornment position="end">
+              <DeleteButton
+                onClick={() => {
+                  onChange(
+                    deleteRelationship(
+                      allRelationships,
+                      currentCategory,
+                      relationship,
+                    ),
+                  );
+                }}
+              />
+            </InputAdornment>
+          ),
+        }}
+      />
+    </div>
+  );
+}
+
+function renderSingleRelationshipObj(
+  allRelationships,
+  targetKey,
+  onChange,
+) {
+  const relationships = get(allRelationships, targetKey, []);
+  return (
+    <div style={{ marginTop: 10 }} key={targetKey}>
+      <TextInput
+        width={240}
+        schema={targetKey}
+        onChange={newName => {
+          onChange(
+            updateCategoryName(allRelationships, targetKey, newName),
+          );
+        }}
+        value={targetKey}
+        autoFocus
+        InputProps={{
+          endAdornment: (
+            <InputAdornment position="end">
+              <IconButton
+                size="small"
+                onClick={() => {
+                  onChange(
+                    addRelationshipToCategory(
+                      allRelationships,
+                      targetKey,
+                      '',
+                    ),
+                  );
+                }}
+              >
+                <NewChildIcon />
+              </IconButton>
+              <DeleteButton
+                onClick={() => {
+                  onChange(
+                    deleteCategory(allRelationships, targetKey),
+                  );
+                }}
+              />
+            </InputAdornment>
+          ),
+        }}
+      />
+      {relationships.map(relationship =>
+        renderSingleRelationship(
```

Comment:
`renderSingleRelationship` should be a component and should be called as one.
Name ideas include RelationshipRole or RelationshipRoleEditor.